modules/javafx.graphics/src/test/java/test/javafx/scene/input/MouseEventTest.java

```diff
@@ -111,6 +111,38 @@
         assertFalse(e.isStillSincePress());
     }
 
+    @Test public void testToStringMatchingBrackets() {
+        Rectangle node = new Rectangle();
+        node.setTranslateX(3);
+        node.setTranslateY(2);
+        node.setTranslateZ(50);
+
+        PickResult pickRes = new PickResult(node, new Point3D(15, 25, 100), 33);
+
+        MouseEvent e = new MouseEvent(MouseEvent.MOUSE_PRESSED,
+                10, 20, 30, 40, MouseButton.PRIMARY, 1,
+                false, false, false, false,
+                true, false, false,
+                false, false, false, pickRes);
+
+        assertSame(pickRes, e.getPickResult());
+
+        String str = e.toString();
+        int bracketCount = 0;
+        for (int i = 0; i < str.length(); i++) {
+            switch (str.charAt(i)) {
+                case '[':
+                    ++bracketCount;
+                    break;
+                case ']':
+                    --bracketCount;
+                    assertTrue("Too many closing brackets: " + str, bracketCount >= 0);
```

Comment:
This test can fail due to a malformed `toString` result in the node (`getIntersectedNode()`), which I would think is outside the scope of this test. In practice, this test's result is dependent on what node I choose to test.

Shouldn't we be testing the structure of the string and not its contents?

Reply:
On the one hand, I can see your point that it's somewhat tangential that Rectangle has matched brackets, but you could say the same about the Point3D. Here is the result of MouseEvent.toString for the test:

```
MouseEvent [source = javafx.event.Event$$Lambda$110/0x0000000800c52fa8@461111c0,
target = javafx.event.Event$$Lambda$110/0x0000000800c52fa8@461111c0, eventType = MOUSE_PRESSED,
consumed = false, x = 18.0, y = 27.0, z = 150.0, button = PRIMARY, primaryButtonDown,
pickResult = PickResult [node = Rectangle[x=0.0, y=0.0, width=0.0, height=0.0, fill=0x000000ff],
point = Point3D [x = 15.0, y = 25.0, z = 100.0], distance = 33.0]]
```

The test would be more complicated if it were changed to ignore the results of the `toString()` of both the  intersected node and the point. If I were to go down this path, I would likely just change it to do match a regex of `"^MouseEvent \[.*PickResult \[.*\]\]$"`, which would be simple. Do you think it's worth it?

Maybe instead I could add a comment that this test checks for matching brackets in the entire composed string returned by `MouseEvent::toString`, including `PickResult::toString`, which in turn includes `Node::toString` for the intersected node?

Reply:
I don't think it matters too much, but a change in any of the classes used here could break this test, which is a bit off. I will be satisfied with a comment warning about this case so if it breaks it will be easy to see where the problem is.